### PR TITLE
chore(thermocycler-gen2): set up channel testing script

### PR DESCRIPTION
**Pull request by fsinapi**
This adds a self contained python testing script intended to test the performance of each individual channel on a Thermocycler.

Other misc changes:
- Added a command in `test_utils` to get the power for each peltier
- Added power data in a separate plot to the `plot_data` tool

Tested by performing a run on an EVT unit

### CODE REVIEW

File: stm32-modules/thermocycler-gen2/scripts/channel_test.py [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+        the first Thermocycler connected over USB.
+        '''
+        self.ser = build_serial(port)
+        self.debug = debug
+    
+    
+    _POLL_FREQ = 2.0 
+    '''Poll frequency in Hz'''
+
+    class Thermistors():
```
Comment (amitlissack) [inline stm32-modules/thermocycler-gen2/scripts/channel_test.py]: The `@dataclass` decorator is a nice tool for these kinds of data classes.

File: stm32-modules/thermocycler-gen2/scripts/channel_test.py [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+        avail = list(grep('.*hermocycler*'))
+        if not avail:
+            raise RuntimeError("could not find thermocycler")
+        return serial.Serial(avail[0].device, 115200)
+    return serial.Serial(port, 115200)
+
+
+class Thermocycler():
+    def __init__(self, port: str = None, debug: bool = False):
+        '''
```
Comment (amitlissack) [inline stm32-modules/thermocycler-gen2/scripts/channel_test.py]: Double quotes for docstrings

**Review by amitlissack — COMMENTED** [review]
Just some little pythonic comments.

**Review by sfoster1 — APPROVED** [review]
Looks good to me!

**Pull request merged**